meadow-client/src/Semantics.purs

```diff
@@ -0,0 +1,1067 @@
+showContract (Let letLabel contract1 contract2) =
+  joinWith "" ["Let ",
+               toString letLabel, " (",
+               showContract contract1, ") (",
+               showContract contract2, ")"]
+showContract (Use letLabel) =
+  joinWith "" ["Use ",
+               toString letLabel]
+
+-- Data type for Inputs with their information 
+data Input = IChoice IdChoice Choice
+           | IOracle IdOracle BlockNumber BigInt
+derive instance eqInput :: Eq Input  
+derive instance ordInput :: Ord Input  
+
+data AnyInput = Action IdAction
+              | Input Input
+derive instance eqAnyInput :: Eq AnyInput  
+derive instance ordAnyInput :: Ord AnyInput  
+
+data IdInput = IdChoice IdChoice
+             | IdOracle IdOracle
+derive instance eqIdInput :: Eq IdInput  
+derive instance ordIdInput :: Ord IdInput  
+
+
+type TimeoutData = M.Map Timeout (S.Set IdCommit)
+
+type CommitInfoRecord = { person :: Person
+                        , amount :: BigInt
+                        , timeout :: Timeout }
+
+data CommitInfo = CommitInfo { redeemedPerPerson :: M.Map Person BigInt
+                             , currentCommitsById :: M.Map IdCommit CommitInfoRecord 
+                             , expiredCommitIds :: S.Set IdCommit
+                             , timeoutData :: TimeoutData }
+--derive instance eqCommitInfo :: Eq CommitInfo  
+--derive instance ordCommitInfo :: Ord CommitInfo  
+
+type OracleDataPoint = { blockNumber :: BlockNumber
+                       , value :: BigInt }
+
+data State = State { commits :: CommitInfo
+                   , choices :: M.Map WIdChoice Choice
+                   , oracles :: M.Map IdOracle OracleDataPoint
+                   , usedIds :: S.Set IdAction}
+--derive instance eqState :: Eq State  
+--derive instance ordState :: Ord State  
+
+-- Adds a commit identifier to the timeout data map 
+addToCommByTim :: Timeout -> IdCommit -> TimeoutData -> TimeoutData 
+addToCommByTim timeout idCommit timData =
+  case M.lookup timeout timData of
+    Just commSet -> M.insert timeout (S.insert idCommit commSet) timData
+    Nothing -> M.insert timeout (S.singleton idCommit) timData
+
+-- Remove a commit identifier from the timeout data map 
+removeFromCommByTim :: Timeout -> IdCommit -> TimeoutData -> TimeoutData
+removeFromCommByTim timeout idCommit timData =
+  case M.lookup timeout timData of
+    Just commSet -> let newCommSet = S.delete idCommit commSet in
+                    if S.isEmpty newCommSet
+                    then M.delete timeout timData
+                    else M.insert timeout newCommSet timData
+    Nothing -> timData 
+
+-- Add a commit to CommitInfo
+addCommit :: IdCommit -> Person -> BigInt -> Timeout -> State -> State
+addCommit idCommit person value timeout (State state) =
+  State (state { commits = CommitInfo (ci { currentCommitsById =
+                                      M.insert idCommit
+                                               ({ person: person
```

Comment:
If the record name matches the variable name, you can leave it out. So `{ person: person }` becomes `{ person }`. Similarly, timeout, and `value` if you just rename it to `amount` in line 273.

Reply:
Ah, thanks, good to know. I think I can leave it in here, since for readability it is almost best, but for future code I will use it.